### PR TITLE
BSP-78: Transformation Rules

### PR DESCRIPTION
**Pull request by iichr**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BSP-78

### Change description ###
Implement basic transformation logic for a subset of example field so that received Excela JSON data is translated/transformed into the appropriate data formats that CCD is expecting for Financial Remedy.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformer/D8FormToCaseTransformer.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+        return map;
+    }
+
+    @Override
+    public Map<String, Object> toCase(ExceptionRecord exceptionRecord) {
+        Map<String, Object> resultingCaseData = new HashMap<>();
+
+        List<OcrDataField> allOcrKeyValuePairs = exceptionRecord.getOcrDataFields();
+        allOcrKeyValuePairs.forEach((erEntry) ->
```
Comment (jvictorcf) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformer/D8FormToCaseTransformer.java]: You can use collect(Collections.toMap()).
Have a look into this: https://github.com/hmcts/div-case-orchestration-service/pull/595/files#diff-d191dfbe31223ba66f4b7c85d4c7f29cR25

Comment (iichr) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformer/D8FormToCaseTransformer.java]: Great idea, much more concise!

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/bulkscan/TransformationController.java [hunk 2]
```diff
@@ -12,16 +12,23 @@
...
+import java.util.Map;
 
 import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
 
 @Slf4j
 @RestController
 public class TransformationController {
 
+    private static final String CASE_TYPE_ID = "FINANCIAL_REMEDY";
+    private static final String EVENT_ID = "EVENT_ID";
```
Comment (jvictorcf) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/bulkscan/TransformationController.java]: Why do we need these values?

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformer/ExceptionRecordToCaseTransformer.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.service.bulkscan.transformer;
+
+import org.mapstruct.Mapping;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.bulkscan.transformation.in.ExceptionRecord;
+
+import java.util.Map;
+
+public interface ExceptionRecordToCaseTransformer {
+    Map<String, Object> toCase(ExceptionRecord exceptionRecord);
```
Comment (jvictorcf) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformer/ExceptionRecordToCaseTransformer.java]: Maybe a more descriptive name?
Please have a look into the Divorce review: https://github.com/hmcts/div-case-orchestration-service/pull/596

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/TransformationControllerTest.java [hunk 4]
```diff
@@ -13,33 +13,42 @@
...
 @WebMvcTest(TransformationController.class)
 public class TransformationControllerTest extends BaseControllerTest {
+
+    private static final String EXCEPTION_RECORD_JSON_PATH = "/fixtures/model/exception-record.json";
     private static final String API_URL = "/transform-exception-record";
     ObjectMapper objectMapper = new ObjectMapper();
 
     @Test
     public void shouldReturnSuccessResponseForTransformationEndpoint() throws Exception {
         JsonNode formToValidate = objectMapper.readTree(new File(getClass()
```
Comment (jvictorcf) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/TransformationControllerTest.java]: Divorce review

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/D8FormToCaseTransformerTest.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+public class D8FormToCaseTransformerTest {
```
Comment (jvictorcf) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/D8FormToCaseTransformerTest.java]: I'm ok with this, but it looks a bit different from Divorce, doesn't it? Maybe I'm imagining it.
If it does, why does it?

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/bulkscan/TransformationController.java (hunk 2 above)
Comment (johnoneill46) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/bulkscan/TransformationController.java]: These are expected in the response to BSP. At the moment we have no idea what the "real" values actually are, but these should suffice for the POC while we wait to find out what the actual values should be

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformer/ExceptionRecordToCaseTransformer.java (hunk 3 above)
Comment (iichr) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformer/ExceptionRecordToCaseTransformer.java]: transformIntoCaseData as on Divorce

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/TransformationControllerTest.java (hunk 4 above)
Comment (iichr) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/TransformationControllerTest.java]: This seems to be the way it's done everywhere in FR, the `ResourceLoader` we know and love form divorce is not used anywhere

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/D8FormToCaseTransformerTest.java (hunk 5 above)
Comment (iichr) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/D8FormToCaseTransformerTest.java]: Renamed both class and test to `FRFormToCaseTransformer`

**Pull request merged**